src/app/controllers/revGraphView.js

```diff
@@ -0,0 +1,1044 @@
+                  var newNodeOpacity = nodActive ? 0 : realNodeOpacity;
+                  var nodData = nod.__data__;
+                  if (nodData == undefined) {continue;};
+                  // Hide or show the elements
+                  nod.getElementsByTagName('use')[0].style.opacity = newNodeOpacity;
+                  // Update the real node opacity
+                  nod.origNodeOpacity = realNodeOpacity;
+                  // Update whether or not the elements are active
+                  nod.active = nodActive;
+
+                  // Hide/show also related links
+                  var linkList = link._groups[0];
+                  var lin;
+                  for (lin of linkList) {
+                    var linkData = lin.__data__;
+                    if ((linkData.source.index == nodData.index) ||
+                        (linkData.target.index == nodData.index)) {
+                      // Check and store the original opacity of a link
+                      var origLinkOpacity = lin.origLinkOpacity ? false : true;
+                      var realLinkOpacity = origLinkOpacity ? lin.attributes['stroke-opacity']['value'] : lin.origLinkOpacity;
+                      // Determine if current link is visible
+                      var linkActive = lin.active ? false : true;
+                      var newLinkOpacity;
+                      if (linkActive) {
+                        newLinkOpacity = 0;
+                      }
+                      else {
+                          var sourceNode = nodeList.filter(nod => nod.__data__.index == linkData.source.index);
+                          var targetNode = nodeList.filter(nod => nod.__data__.index == linkData.target.index);
+                          var sourceNodeOpacity = sourceNode[0].active ? false : true;
+                          var targetNodeOpacity = targetNode[0].active ? false : true;
+                          if ((sourceNodeOpacity == true) && (targetNodeOpacity == true))
+                            {
+                            newLinkOpacity = realLinkOpacity;
+                          }
+                          else {
+                            newLinkOpacity = 0;
+                          }
+                      }
+                      // Hide or show the elements
+                      lin.attributes['stroke-opacity']['value'] = newLinkOpacity;
+                      // Update the real node opacity
+                      lin.origLinkOpacity = realLinkOpacity;
+                      // Update whether or not the elements are active
+                      lin.active = linkActive;
+                      // Update also the middle arrow
+                      if (lin.attributes['stroke-opacity']['value'] == 0) {
+                        lin.attributes['marker-mid']['value'] = ""
+                      }
+                      else {
+                        lin.attributes['marker-mid']['value'] = "url(#arrow)"
+                      }
```

Comment:
Same as for the nodes. Ideally, you'd want to record that a nodeFilter has been (de-selected) and use this to update your **model** (the graph). E.g. by setting some flag. Angular/HTML/CSS should then take care of updating the opacity of the svg/html elements based on a single rule/function which calculates the opacity of nodes in the graph based on the properties of the nodes/links.

Reply:
Refactored following these suggestions.